G5.195 Updating testing grid and adding support for multiple versions of PHPUnit

vidu-wil opened the pull request:
**Issue #195**
**Issue #200**

## Motivation

<!-- This can usually be copied from the issue. Please do not just say, go see issue but instead copy the relevant details here. -->
We need to add support for Drupal 10.6, 11.3 and PHP 8.4, 8.5 as well as remove support for Drupal 10.4, 11.1, and PHP 8.1 and update the automated testing grid to test these versions. Also, we currently have a phpunit.xml but only one. We want to update to having a phpunit.9.xml, phpunit.10.xml and phpunit.11.xml`. 

## What does this PR do?
<!-- *Please describe each things this PR does. For example, a PR may 1) solve a specific bug, 2) create an automated test to ensure it doesn't return.* -->

- [x] Adds support for PHP 8.4, 8.5, Drupal 10.6 and Drupal 11.3
- [x] Removes support for Drupal 10.4, Drupal 11.1 and PHP 8.1
- [x] Adds support for PHPUnit versions 9, 10 and 11
- [x] Adds support for PostgreSQL 18 
- [x] Removes support for PostgreSQL 13
- [x] Schedules each test grid to run at 2 am instead of PHPUnit 

## Testing

<!-- ### Automated Testing
*Please describe each automated test this PR creates and provide a list of the assertions it makes using casual language.*
*Do not just say things like "asserts the array is not empty" but rather say "Ensures that the return value of method X with these parameters is not an empty array".* -->


### Manual Testing
<!-- *Describe in detail how someone should manually test this functionality.*
*Make sure to include whether they need to build a docker from scratch, create any records, etc.* -->

1. Checkout this branch locally, open it in VSCode and "Reopen in Container".
2. Confirm that devcontainer builds properly and open the webpage in the browser to confirm versions. Note it is now building Drupal 11.3 and PHP 8.5
<img width="540" height="135" alt="Screenshot 2025-12-24 at 10 05 58 AM" src="https://github.com/user-attachments/assets/b0a8f264-f7f4-474d-a514-efee3bb649dd" />

3. Check the Automated workflows that they are being run (and passing!) on:
    - PHP 8.2, 8.3, 8.4 & 8.5 but NOT PHP 8.1.
    - PostgreSQL 14 & 18
    - Drupal 10.5, 10.6, 11.x, 11.2 and 11.3, but NOT Drupal 10.4 or 11.1.
 
<img width="624" height="411" alt="Screenshot 2026-01-21 at 9 23 46 AM" src="https://github.com/user-attachments/assets/cf1960b0-84bb-41bf-8728-e84fb0028c32" />

4. Login to the site built by this container and check that it is on the latest version of everything (Drupal 11.3-dev, PHP 8.5.0, PostgreSQL 18.1) at Admin > Reports > Status report
5. Check the README on this branch and confirm the testing grid renders properly with its badges

To test whether the phpunit tests using the correct phpunit.xml file:
1. Change the drupal, php and pgsql versions on `.devcontainer.json` and rebuild the container.
2. Check if the phpunit.xml file is being updated locally to the correct version of phpunit.
3. Run the phpunit tests to confirm if they're working correctly.

vidu-wil committed:
added phpversions 8.4, 8.5 , drupal 10.6, 11.3 support and removed drupal 10.4 and 11.1 support

vidu-wil committed:
updated the postgresql version support

vidu-wil committed:
changed the drupal version to bleeding edge and updated the php and pgsql versions

vidu-wil committed:
updated test-tripal-action version

vidu-wil committed:
added different versions of phpunit.xml files

vidu-wil committed:
removed listeners element as they are no longer supported

vidu-wil committed:
Updated all workflows using actions/checkout to version 6

vidu-wil committed:
linked the phpunit.xml with the correct version of phpunit file

vidu-wil committed:
added init.sh back on postCreateCommand

vidu-wil committed:
added postgresql 18 support

vidu-wil committed:
excluded postgresql 18 from middle drupal versions

reynoldtan commented:
Executed the manual test instructions and confirmed the following: 

- [x] Container builds properly with the correct version
<img width="375" height="109" alt="Screen Shot 2026-01-08 at 1 37 10 PM" src="https://github.com/user-attachments/assets/000ba4cc-fa45-44e8-a80b-f8bc9876b057" />

- [ ] Passing for 
    
PHP 8.1, 8.2, 8.3, 8.4 & 8.5
PostgreSQL 13 & 18
Drupal 10.5, 10.6, **11.x**, 11.2 and 11.3, but NOT Drupal 10.4 or 11.1.

No Drupal **11.x** entry in the Checks list below

<img width="586" height="431" alt="Screen Shot 2026-01-08 at 1 42 33 PM" src="https://github.com/user-attachments/assets/b83f70e5-a97a-4d2c-a02c-1090a1e908e8" />


- [x] Software stack in the latest versions
<img width="565" height="363" alt="Screen Shot 2026-01-08 at 1 47 35 PM" src="https://github.com/user-attachments/assets/c749064f-6e40-4184-97c6-03c089fa854f" />

- [ ] Grid renders with badges.
<img width="833" height="248" alt="Screen Shot 2026-01-08 at 1 51 20 PM" src="https://github.com/user-attachments/assets/484072e1-b1af-4996-8c31-3ba8a7f60f3c" />

Links go to a page not found

- [ ] PHPUnit test
I changed the version to

```
  "build": {
    "dockerfile": "Dockerfile",
    "args": {
      "drupalversion": "10.5-dev",
      "phpversion": "8.1",
      "pgsqlversion": "13"
    }
  },
```
<img width="1094" height="1256" alt="image" src="https://github.com/user-attachments/assets/0a658679-59f3-43db-98eb-638d0bf61a83" />

laceysanderson commented on .github/workflows/MAIN-phpunit-php8.4_D10_5x.yml:
```diff
@@ -0,0 +1,31 @@
+  workflow_dispatch:
+  push:
+    branches:
+      - 4.x
```
To fix the badges not rendering you'll want to add this branch to these "MAIN" files so it runs. This allows us to confirm these workflows work as intended before merging and allows us to confirm the badges.

vidu-wil committed:
added the current branch on the workflow MAIN files for the badges to update

vidu-wil committed:
removed postgresql 13 support

vidu-wil commented:
> * [ ]  PHPUnit test
>   I changed the version to
> 
> ```
>   "build": {
>     "dockerfile": "Dockerfile",
>     "args": {
>       "drupalversion": "10.5-dev",
>       "phpversion": "8.1",
>       "pgsqlversion": "13"
>     }
>   },
> ```
> 
> <img alt="image" width="1094" height="1256" src="https://private-user-images.githubusercontent.com/15472253/533561091-0a658679-59f3-43db-98eb-638d0bf61a83.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjgyNDUxMjgsIm5iZiI6MTc2ODI0NDgyOCwicGF0aCI6Ii8xNTQ3MjI1My81MzM1NjEwOTEtMGE2NTg2NzktNTlmMy00M2RiLTk4ZWItNjM4ZDBiZjYxYTgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTEyVDE5MDcwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMxYzYwMTYwMmQ2YTJkNjU3NzgzZGJmN2I0ZjQxNjZmNWIyZmU1MDRmYzk4OTU3MDU2MTg2NzdmNWQwOTUxN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4vOvPogJr1BRcBCOd9U4zbL40r9Ab7NmMwKE1ezjAgw">

@reynoldtan Thank you for catching this error. I have now removed support for PostgreSQL13 and the earliest version that this module supports is PostgreSQL14. You should be able to build the docker container for that version and it should work as expected now.

vidu-wil committed:
removed php 8.1 support, added drupal 10.5+php8.4 and
drupal 10.6+php8.4 combination, replaced pgsqlversion with postgresqlversion

vidu-wil committed:
matched the test versions and docker builds to tripal

vidu-wil committed:
scheduled each grid to run at 2am instead of ALL-PHPUnit

laceysanderson committed:
Merge branch '4.x' into g5.195-updateTestingGrid

laceysanderson commented on .github/workflows/ALL-PHPUnit.yml:
```diff
@@ -26,34 +21,52 @@ jobs:
+          - php-version: '8.4'
+            drupal-version: '10.5.x-dev'
+          - php-version: '8.4'
+            drupal-version: '10.6.x-dev'
```
Needs to be removed.

laceysanderson commented on .github/workflows/ALL-PHPUnit.yml:
```diff
@@ -26,34 +21,52 @@ jobs:
+            drupal-version: "11.3.x-dev"
+          # Exclude higher postgresql version from middle Drupal versions
+          - pgsql-version: "18"
+            drupal-version: "10.5.x-dev"
```
Needs to be 10.6

laceysanderson committed:
Merge branch '4.x' into g5.195-updateTestingGrid

laceysanderson committed:
Fix changes mentioned in PR review.

laceysanderson committed:
Merge branch 'g5.195-updateTestingGrid' of https://github.com/TripalCultivate/TripalCultivate-Phenotypes into g5.195-updateTestingGrid

laceysanderson reviewed (approved):
Perfect ✅
